### DISCUSSION TITLE
Reversing a SourceList?

### DISCUSSION THREAD

**Discussion by houstonhaynes** [JordanMarr/ReactiveElmish.Avalonia#40] [Q&A] [answered]
I feel like I've asked this before but I can't find it here - about showing info in a sourcelist with the "latest" data at the top. The challenge is that even with a sortable view in the datagrid control, it "resets" when new data arrives. So the list gets re-ordered. When I was managing the list myself in the earlier version of the "Chart" series Actions list I was reversing, adding, then reversing again before pushing to the model. So it was "easy" so to speak. 

I looked at the DynamicData docs and see that they contemplate re-ordering - among other types of manipulations - but I'm not sure if there's a way for us to use it it with the wrapper in its current form. Thanks!

**Comment by JordanMarr** [comment]
Hmm. Maybe it would be better to Swap out with the `ObservableCollection` instead.  

if that works, then i might add a similar helper module around `ObservableCollection`.

**Reply by houstonhaynes** [reply]
I'll give it a spin - should be easy enough to revert to since it's already in another spot

**Comment by JordanMarr** [comment, marked as answer]
After actually trying it, I am unable to reproduce your issue. 
The sample CounterView is using `Actions: SourceList<Action>`, and I am able to sort the table:

```F#
member this.Actions = this.BindSourceList(local.Model.Actions)
```

![image](https://github.com/JordanMarr/ReactiveElmish.Avalonia/assets/1030435/320d4c51-5c6d-48a0-afa4-760021eba366)
![image](https://github.com/JordanMarr/ReactiveElmish.Avalonia/assets/1030435/fd0b62b3-a8c5-4178-b2ce-4eca349730d5)

**Reply by houstonhaynes** [reply]
Not sure what I was seeing before (or what state things were in that the sort was getting cleared on update of the MVU loop) but you're right - please disregard!
